src/Core/Pricing/PriceEngine.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Pricing;
+
+use PrestaShop\Decimal\DecimalNumber;
+use PrestaShop\PrestaShop\Core\Pricing\Extension\PriceExtensionInterface;
+
+class PriceEngine
+{
+    private array $listeners = [];
+
+    public function __construct(
+        private array $extensions,
+    ) {
+        foreach ($this->extensions as $extension) {
+            $this->addExtension($extension);
+        }
+    }
+
+    public function getPrice(int $productId, array $context = []): Price
+    {
+        $quantity = 1; // todo, find what do to with this. Is this should live inside the context or as an argument for example.
+
+        $context += [
+            'productId' => $productId,
+        ];
+
+        $context['quantity'] = $context['quantity'] ?? $quantity;
+
+        $priceHT = new Price(new DecimalNumber('0'));
```

Comment:
Note for future: I think HT means "Hors Taxe" for you and that is French. If we write it in English, we need "TE" or "Tax excluded" 😉

Reply:
Sorry about that, will be fixed

Reply:
🥐